### PR TITLE
fix(robot-server): tip length cal -  use nozzle height for tip length calculation

### PR DESCRIPTION
**Pull request by ahiuchingau**
# Overview
In a previous refactor, the height of the pipette tip is incorrectly used in calculating the tip length during tip length calibration. This PR fixes the issue by passing the correct critical point to get the robot's gantry position.

# Changelog
- pull `get_current_point()` from all calibration flows to util.py
- add optional critical point arg to `get_current_point()`

# Review requests
1. Run tip length calibration to make sure the tip length data look correct (10 uL/20 uL tip length should be around 30 mm)
2. Make sure pipettes are moving to the correct locations in all calibration flows 

# Risk assessment
Medium, a refactor that touches all three cal flows but the change itself is pretty small

### CODE REVIEW

**Comment by ahiuchingau** [comment]
For some reason, even though the `critical_point` argument is missing from all `get_current_points()` calls in the tests, no mypy errors are raised (similar to the bug introduced in #6505)

**Review by sfoster1 — APPROVED** [review]
Makes sense!

File: robot-server/robot_server/robot/calibration/util.py [hunk 1]
```diff
@@ -79,6 +79,13 @@ def get_next_state(self, from_state, command):
     'PipetteOffsetCalibrationUserFlow']
 
 
+async def get_current_point(
```
Comment (sfoster1) [inline robot-server/robot_server/robot/calibration/util.py:82]: On a second reading actually, I don't think pulling `get_current_point` out from the user flows into this util is a good idea, for a couple reasons:

1. It's now depending on the "private" (underscore-prefixed) member `_hardware` of `CalibrationUserFlow`, which by convention isn't actually part of the type contract of `CalibrationUserFlow`
2. Calling a pure function and passing `self` is a pretty big code smell
3. This function is a one-liner that delegates to a different function without even really changing the arguments

Put all that together and I think things would stay a lot nicer and simpler just changing the callsites and not factoring this out. I mean, all the user flows still have `get_current_point` functions that just delegate to this one after all.

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Comments inline - I think splitting `get_current_point` out into a util file just adds some unnecessary complication

File: robot-server/robot_server/robot/calibration/util.py (hunk 1 above)
Comment (ahiuchingau) [reply robot-server/robot_server/robot/calibration/util.py:82]: Yea that makes sense to me!

**Review by sfoster1 — APPROVED** [review]
LGTM!

**Review by b-cooper — APPROVED** [review]
Tested on robot, works as expected

**Pull request merged**